### PR TITLE
Introduce `Index` and `MultiIndex`.

### LINKED ISSUE #306: Support more than one index for the index property
Both DataFrame and Series have this limitation:

```python
    @property
    def index(self):
        """The index (row labels) Column of the DataFrame.

        Currently supported only when the DataFrame has a single index.
        """
        from databricks.koalas.series import Series
        if len(self._metadata.index_map) != 1:
            raise KeyError('Currently supported only when the DataFrame has a single index.')
        return Series(self._index_columns[0], anchor=self, index=[])
```

We should just support more than 1 here ... (and also 0).

### PR DESCRIPTION
**Pull request by ueshin**
This PR introduces `Index` and `MultiIndex` to behave similar to pandas' indexes.
Resolves #306.

### CODE REVIEW

**Comment by ueshin** [comment]
I'll add docs later.

**Comment by ueshin** [comment]
@rxin I agree that we can reuse many of code in `Series`, but there are still many differences between `Index` and `Series` according to my rough investigation.

no. of common functions | no. of only in Series | no. of only in Index
--------------- | -------------------- | ------------------
49 | 131 | 39

Note that even the common functions here might behave differently between `Series` and `Index`.

Also, I'm feeling weird if we return `Series` for a single index whereas `MultiIndex` for more than one index since they are hierarchically different.

How about introducing a mix-in for common functions like `IndexOpsMixin` (from pandas' name) or something to reuse code?

**Comment by rxin** [comment]
Sgtm

**Comment by ueshin** [comment]
@rxin Thanks! I will address them soon.

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -972,6 +972,10 @@ def index(self):
         """The index (row labels) Column of the DataFrame.
 
         Currently not supported when the DataFrame has no index.
```
Comment (rxin) [inline databricks/koalas/frame.py]: maybe we should return something ... like None?

**Comment by rxin** [comment]
@HyukjinKwon @ueshin can you review each other's code? I will be gone for almost 2 wks :)

**Comment by HyukjinKwon** [comment]
Oops, will do. sry.

**Comment by HyukjinKwon** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
For https://github.com/databricks/koalas/pull/341#issuecomment-492917885, seems we (I) have added `any` and `all`, for instnace. let me try to put it in the IndexOpsMixin too.

**Pull request merged**